tools/auth/wakatime.go

```diff
@@ -0,0 +1,83 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pocketbase/pocketbase/tools/types"
+	"golang.org/x/oauth2"
+)
+
+func init() {
+	Providers[NameWakatime] = wrapFactory(NewWakatimeProvider)
+}
+
+var _ Provider = (*Wakatime)(nil)
+
+// NameWakatime is the unique name of the Wakatime provider.
+const NameWakatime = "wakatime"
+
+// Wakatime is an auth provider for Wakatime.
+type Wakatime struct {
+	BaseProvider
+}
+
+// NewWakatimeProvider creates a new Wakatime provider instance with some defaults.
+func NewWakatimeProvider() *Wakatime {
+	return &Wakatime{BaseProvider{
+		ctx:         context.Background(),
+		displayName: "Wakatime",
+		pkce:        false,
+		scopes:      []string{"read_summaries","email"},
```

Comment:
Let's leave the minimal scopes for now, aka. just the `email` because the provider also seems to suggest specifying only the summaries that are needed:

> Consider instead requesting scopes for only the summaries you need. For ex: scope=read_summaries.languages,read_summaries.editors to only request access to the user’s language and editor summaries and stats.

Users can specify these and any other scopes from the client-side with the `authWithOAuth2` call.